CHANGELOG.md

```diff
@@ -8,6 +8,7 @@
 
 ### Fixed
 * Allow defining properties with the field name as the persisted name. ([#1240](https://github.com/realm/realm-kotlin/issues/1240))
+* Fix compilation error when accessing Realm Kotlin model classed from Java code. ([#1256](https://github.com/realm/realm-kotlin/issues/1256))
```

Comment:
Can we add an integration test for this? I assume adding a Java class to one of our JVM integration samples should work?

Reply:
I tried adding a plain Java class as part of the test suite and it didn't trigger the issue as it did in my reproduction project 🤔

Reply:
Ahh. I did it in the `jvmTest`. Will try in the Android tests instead.

Reply:
I managed to trigger it in the `integration-tests/gradle-plugin-test/single-platform` with https://github.com/realm/realm-kotlin/pull/1259/files#diff-343ba0d05e9a620f83dfed66a489bb4016ccade98fe3984c6fb5a7eacf66bdf6R1. Apparently it is only an issue if the accessing Java class is in the `java` source directory 🤯. Stopped investigating options to put it inside `test-base` as I didn't want to pollute our tests by adding the additional source setup, but could try to investigate if we rather would like it to be here?

Reply:
I think having it in an integration test is just fine 👍